iogt_content_migration/management/commands/load_v1_users.py

```diff
@@ -0,0 +1,365 @@
+
+    def migrate(self):
+        self.populate_content_type_map()
+
+        self.migrate_user_groups()
+        self.migrate_user_accounts()
+        self.mark_user_registration_survey_required()
+
+        self.migrate_user_comments()
+        self.migrate_user_survey_submissions()
+        self.migrate_user_poll_submissions()
+        self.migrate_user_freetext_poll_submissions()
+
+    def populate_content_type_map(self):
+        sql = f'select * from django_content_type'
+        cur = self.db_query(sql)
+
+        for row in cur:
+            new_content_type = ContentType.objects.filter(model=row["model"]).first()
+            if not new_content_type:
+                self.stdout.write(self.style.ERROR(f'Content Type missing for {row["model"]}'))
+            self.content_type_map.update({
+                row["model"]: new_content_type
+            })
+
+        self.content_type_map.update({
+            'articlepage': ContentType.objects.filter(app_label='home', model='article').first()
+        })
+
+        cur.close()
+
+    def migrate_user_accounts(self):
+        sql = f'select * from auth_user'
+        cur = self.db_query(sql)
+
+        for row in self.with_progress(sql, cur, 'User Migration in progress'):
+            v1_user_id = row.pop('id')
+
+            user_data = dict(row)
+            user_data.update({'has_filled_registration_survey': True})
+
+            user, __ = get_user_model().objects.update_or_create(username=row['username'], defaults=user_data)
+            V1ToV2ObjectMap.create_map(user, v1_user_id)
+
+            user_groups_sql = f'select * from auth_user_groups aug ' \
+                              f'inner join auth_group ag on aug.group_id = ag.id ' \
+                              f'where user_id={v1_user_id}'
+
+            user_groups_cursor = self.db_query(user_groups_sql)
+
+            user.groups.clear()
+            for row_ in user_groups_cursor:
+                group = Group.objects.get(name=row_['name'])
+                group.user_set.add(user)
+
+            user_groups_cursor.close()
+
+    def migrate_user_groups(self):
+        self.stdout.write(self.style.SUCCESS('Starting User Groups Migration'))
+
+        sql = f'select * from auth_group'
+        cur = self.db_query(sql)
+
+        for row in cur:
+            group, __ = Group.objects.get_or_create(name=row['name'])
+            V1ToV2ObjectMap.create_map(group, row['id'])
+
+        self.stdout.write(self.style.SUCCESS('Completed User Groups Migration'))
+
+    def mark_user_registration_survey_required(self):
+        users = get_user_model().objects.filter(groups__id__in=self.registration_survey_mandatory_group_ids)
+        users.update(has_filled_registration_survey=False)
+
+    def migrate_user_comments(self):
+        self.stdout.write(self.style.SUCCESS('Starting Comment migration'))
+
+        sql = f'select * from django_comments dc inner join django_content_type dct on dc.content_type_id = dct.id'
```

Comment:
It appears that this join has two fields called 'id'. One of them appears to be the comment id, while the other is always 32. When I ran the migration this time, I only got a single comment, because now the id is used to detect duplicates and it was using the id whose value is always 32.